### ISSUE TITLE
default mean should not be from mean_var

### ISSUE THREAD

**Issue by smidl** [nmheim/GenerativeModels.jl#41]
Defining mean from mean_var makes sense only for Gaussian VAE, otherwise, it does not work. I propose to do this for CGaussian with DiagVar, but not for others. Calling  default mean() for UnitVar does not make any sense.

**Comment by nmheim** [comment]
CGaussian with DiagVar and ScalarVar need to call the mapping to get either mean or variance, only UnitVar does not need to call the mapping to get the variance. As UnitVar, as I see it, is the special case here I can change it to this:
```julia
mean(p::CGaussian{T,UnitVar}, z::AbstractArray) = p.mapping(z)
variance(p::CGaussian{T,UnitVar}, z::AbstractArray) = fill!(similar(z, xlength(p)), 1)
mean_var(p::CGaussian{T,UnitVar}, z::AbstractArray) = (mean(p, z), variance(p, z))
```
the variance still needs the input z and the similar to create a CuArray if the model is on the GPU.

**Comment by nmheim** [comment]
we will solve this by moving the UnitVar special case to SharedVarCGaussian. We will also rename like:
```julia
CGaussian -> CMeanVarGaussian
SharedVarCGaussian -> CMeanGaussian
```

**Comment by nmheim** [comment]
closed by b27c26d653d6e31e788ea9ae48ff8f337e1d50f5
